### PR TITLE
Update Npm installs to use local feed

### PR DESCRIPTION
**Pull request by MaddyDev**
Following TSG: https://eng.ms/docs/cloud-ai-platform/azure-edge-platform-aep/aep-engineering-systems/productivity-and-experiences/network-isolation/oss/npm

### CODE REVIEW

File: builds/azure-pipelines/template-steps-build-test.yml [hunk 1]
```diff
@@ -60,6 +65,11 @@ steps:
   displayName: Set logging level
   condition: and(succeeded(), ne(variables['AFSQLEXT_TEST_LOGLEVEL'], ''))
 
+- task: npmAuthenticate@0
+  displayName: 'npm Authenticate for project'
+  inputs:
+  workingFile: $(Build.SourcesDirectory)/.npmrc
+  
 - script: |
     npm install
```
Comment (Charles-Gagnon) [inline builds/azure-pipelines/template-steps-build-test.yml:74]: What about this? (and other ones below)

File: builds/azure-pipelines/template-steps-build-test.yml [hunk 2]
```diff
@@ -19,10 +19,15 @@ steps:
     packageType: sdk
     version: '3.1.x'
 
-- script: npm install -g azure-functions-core-tools
+- task: npmAuthenticate@0
```
Comment (Charles-Gagnon) [inline builds/azure-pipelines/template-steps-build-test.yml:22]: Make sure you check with a buddy build that these are pulling from the right npmrc file in our ADO pipelines.

File: builds/azure-pipelines/template-steps-build-test.yml (hunk 1 above)
Comment (Charles-Gagnon) [reply builds/azure-pipelines/template-steps-build-test.yml]: Oh right, nevermind. Local installs pick up the local config. It's only the global ones that need to specify the config directly.

Comment (MaddyDev) [reply builds/azure-pipelines/template-steps-build-test.yml]: added this las week and forgot to clean up, I just cleaned up and added userconfig for the below ones.

File: builds/azure-pipelines/template-steps-build-test.yml (hunk 2 above)
Comment (MaddyDev) [reply builds/azure-pipelines/template-steps-build-test.yml:22]: yes, am going to test with a buddy build.

**Comment by Charles-Gagnon** [comment]
FYI - for the PR builds to work you're going to need to add a default npmrc file here pointing at npmjs

File: builds/azure-pipelines/template-steps-build-test.yml (hunk 1 above)
Comment (MaddyDev) [reply builds/azure-pipelines/template-steps-build-test.yml]: What I understood is, when using as a per project config, it needs to be a sibling of the package.json but when you're specifying it as userconfig, I think you can give the path to the file. But am not 100% sure either. Do you think if I should be adding a .npmrc file here as well?

Comment (Charles-Gagnon) [reply builds/azure-pipelines/template-steps-build-test.yml]: So the issue we have is that we have multiple projects to build (e.g. samples-js and test-js). So in order for it to pick up the npmrc directly we'd have to have one in each of those folders - it doesn't seem to contain any logic to automatically look in parent folders unfortunately.

What you have seems fine - just setting it at the user level will cause it to use that instead. But yes, you will need to add an npmrc file in order for this to successfully build.

File: .npmrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+registry=https://msdata.pkgs.visualstudio.com/SQLToolsAndLibraries/_packaging/SQLToolsAndLibraries_PublicPackages/npm/registry/
```
Comment (Charles-Gagnon) [inline .npmrc:1]: This needs to point to npmjs.org, NOT our private feed

Comment (Charles-Gagnon) [inline .npmrc:1]: This needs to be pointing to the public registry, NOT our private feed (we shouldn't have anything about our private feeds in this repo)

```suggestion
registry=https://registry.npmjs.org
```

Comment (Charles-Gagnon) [reply .npmrc]: It's fine (and expected) for our PR build to use the public registries. It's only the ADO pipelines that need to use the private stuff.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**